hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/invalidation/BatchInvalidator.java

```diff
@@ -164,11 +172,17 @@ public void stateChanged(LifecycleEvent event) {
         });
     }
 
-    private void startBackgroundBatchProcessor() {
-        ExecutionService executionService = nodeEngine.getExecutionService();
-        executionService.scheduleWithRepetition(invalidationExecutorName,
-                new BatchInvalidationEventSender(), batchFrequencySeconds, batchFrequencySeconds, SECONDS);
+    private void checkBackgroundTaskIsRunning() {
+        if (runningBackgroundTask.get()) {
```

Comment:
I think you can leave out the `runningBackgroundTask.get()`, since `runningBackgroundTask.compareAndSet(false, true)` already covers that.